### PR TITLE
Tolerate missing or different tag in subject

### PR DESCRIPTION
**Pull request by zregvart**
When comparing the subjects given `IMAGE_URL` and `IMAGE_DIGEST` task results with the attestation subject in cases when `IMAGE_URL` contains the image tag, and the attestation subject.name doesn't we still don't want to report a violation if `IMAGE_DIGEST` matches subject.digest.

Given that we're now removing strings from subject and results to compare without tags, we also want to be careful not to compare empty strings and consider them valid. I think.

### CODE REVIEW

**Comment by zregvart** [comment]
I thought it was an issue if IMAGE_URL or IMAGE_DIGEST were empty or missing values, I'm not so certain that can be an issue, I left it in anyways, please chime in if I should remove that.

**Review by joejstuart — APPROVED** [review]
LGTM! I'm sure you verified it, but it's working for me. No violations when I use this policy.

**Comment by zregvart** [comment]
> LGTM! I'm sure you verified it, but it's working for me. No violations when I use this policy.

Tests did it for me :) Let's see what others think though before merging...

File: policy/release/slsa_build_scripted_build.rego [hunk 1]
```diff
@@ -66,7 +66,12 @@ deny contains result if {
 		task_result(build_task, "IMAGE_URL").value,
 		task_result(build_task, "IMAGE_DIGEST").value,
 	])
-	subject_image_ref != result_image_ref
+	subject_image_ref_no_tag := regex.replace(subject_image_ref, ":[^@]+@", "@")
```
[the remark this answers is not in the document]
Comment (zregvart) [reply policy/release/slsa_build_scripted_build.rego:69]: Tots, will refine

**Pull request merged**